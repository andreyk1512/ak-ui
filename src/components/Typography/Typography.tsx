import { TypographyProps } from './Typography.types';

export const Typography = ({ children }: TypographyProps) => <article>{children}</article>;
