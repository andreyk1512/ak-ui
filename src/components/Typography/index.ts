export * from './Typography';
export * from './Typography.types';

import { Text, Title, Paragraph } from './components';
import { Typography as OriginTypography } from './Typography';

export type TypographyProps = typeof OriginTypography & {
  Text: typeof Text;
  Title: typeof Title;
  Paragraph: typeof Paragraph;
};

export const Typography = OriginTypography as TypographyProps;
Typography.Text = Text;
Typography.Title = Title;
Typography.Paragraph = Paragraph;
