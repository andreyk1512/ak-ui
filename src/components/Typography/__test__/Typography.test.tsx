import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Typography } from '../Typography';

describe('Typography', () => {
  it('should render the article text', () => {
    const children = 'Lorem ipsum dolor sit amet...';
    const { container } = render(<Typography>{children}</Typography>);
    const article = screen.getByText(children);

    expect(article).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
