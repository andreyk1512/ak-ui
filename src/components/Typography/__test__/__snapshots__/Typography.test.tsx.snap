// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should correctly render article element with content 1`] = `
<article>
  Lorem ipsum dolor sit amet...
</article>
`;
